mongo/datadog_checks/mongo/mongo.py

```diff
@@ -107,11 +107,21 @@ def __init__(self, name, init_config, instances=None):
             hosts = self.instance.get('hosts', [])
             if not hosts:
                 raise ConfigurationError('No `hosts` specified')
+
+            username = self.instance.get('username')
+            password = self.instance.get('password')
+
+            if username and not password:
+                raise ConfigurationError('`password` must be set when a `username` is specified')
+
+            if password and not username:
+                raise ConfigurationError('`username` must be set when a `password` is specified')
```

Comment:
Let's update the README with the username/password/host/port option examples

Reply:
=> Submitted https://github.com/DataDog/integrations-core/pull/6693